t/33-vagrant.t

```diff
@@ -0,0 +1,511 @@
+#!/usr/bin/perl
+
+use Cwd qw(abs_path);
+use Test::Most;
+use File::chdir;
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '20';
+use Test::MockRandom 'backend::vagrant';
+use Test::MockObject;
+use Test::MockModule;
+use Test::Warnings ':report_warnings';
+use Test::Output qw(stderr_like);
+use backend::vagrant;
+
+use bmwqemu ();
+
+# hide debug messages, keep warnings & errors
+$bmwqemu::logger = Mojo::Log->new(level => 'warn');
+
+my $mock_vagrant = Test::MockModule->new("vagrant", no_auto => 1);
+$mock_vagrant->noop('mkdir');
+
+my $tmpdir        = File::Temp->newdir();
+my $mock_file_tmp = Test::MockModule->new("File::Temp");
+$mock_file_tmp->redefine('newdir', sub { return $tmpdir; });
+
+my $expected_console_name        = undef;
+my $expected_console_type        = undef;
+my %expected_console_credentials = ();
+my $distri                       = Test::MockModule->new('distribution');
+$distri->redefine('add_console', sub {
+        my ($self, $name, $type, $creds) = @_;
+
+        if (defined($expected_console_name)) {
+            is($name, $expected_console_name, 'Check the name of the console');
+        }
+        if (defined($expected_console_type)) {
+            is($type, $expected_console_type, 'Check the type of the console');
+        }
+        if (%expected_console_credentials) {
+            is_deeply($creds, \%expected_console_credentials, 'Check the name of the console');
+        }
+        my $ret = Test::MockObject->new();
+        $ret->set_true('backend');
+        return $ret;
+});
+$testapi::distri = distribution->new;
+
+my $check_socket_ret            = undef;
+my $check_socket_fh_expected    = undef;
+my $check_socket_write_expected = undef;
+my $mock_virt                   = Test::MockModule->new("backend::virt", no_auto => 1);
+$mock_virt->redefine('check_socket', sub {
```

Comment:
backend::virt::check_socket does not exist. Did you forget to commit something?

Reply:
Ups, no, I just did not check where `check_socket` is defined.